### PR TITLE
Deprecated the pre_execute_functions and post_execute_functions

### PR DESCRIPTION
**Pull request by pause125**
## Summary

part of #2188

### CODE REVIEW

File: crates/rooch-genesis/released/test
Comment (jolestar) [inline crates/rooch-genesis/released/test:1]: testnet 的 genesis 不应该改变, 是因为 MoveOSTransaction 的序列化格式变了？ 感觉 MoveOSTransaction 的 pre_execute_function 和 post_execute_function 字段得留着给反序列化用，或者自定义反序列化方法然后特殊处理一下。

Comment (jolestar) [reply crates/rooch-genesis/released/test:1]: 序列化的时候不写入，反序列化的时候兼容就可以

File: moveos/moveos-types/src/transaction.rs [hunk 1]
```diff
@@ -269,21 +265,48 @@ impl MoveOSTransaction {
...
+        D: Deserializer<'de>,
+    {
+        #[derive(::serde::Deserialize)]
+        struct OldValue(
+            (
+                ObjectMeta,
+                TxContext,
+                MoveAction,
+                Option<Vec<FunctionCall>>,
+                Option<Vec<FunctionCall>>,
```
Comment (jolestar) [inline moveos/moveos-types/src/transaction.rs:298]: Should be `Vec<FunctionCall>`, not `Option<Vec<FunctionCall>>`.

File: moveos/moveos-types/src/transaction.rs [hunk 2]
```diff
@@ -269,21 +265,48 @@ impl MoveOSTransaction {
...
+            ),
+        );
 
-    pub fn append_post_execute_functions(&mut self, functions: Vec<FunctionCall>) {
-        self.post_execute_functions.extend(functions);
+        // FIXME: This is a hack to deserialize the old MoveOSTransaction format.
+        // We need compatible deserialization logic to support both old and new formats.
```
[the remark this answers is not in the document]
Comment (jolestar) [reply moveos/moveos-types/src/transaction.rs:303]: Try using OldValue first; if it raises an error, use new formats.

[the remark this answers is not in the document]
Comment (baichuan3) [reply moveos/moveos-types/src/transaction.rs:303]: In another PR, I am solving a similar problem, which requires compatibility between new and old data. The idea is to parse according to struct field by field. There are still some problems to be solved. You can refer to that later.

[the remark this answers is not in the document]
Comment (jolestar) [reply moveos/moveos-types/src/transaction.rs:303]:
```
use serde::{Deserialize, Deserializer};

#[derive(Debug, Clone, Serialize)]
pub struct VerifiedMoveOSTransaction {
    pub root: ObjectMeta,
    pub ctx: TxContext,
    pub action: VerifiedMoveAction,
}

impl<'de> Deserialize<'de> for VerifiedMoveOSTransaction {
    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
    where
        D: Deserializer<'de>,
    {
        #[derive(Deserialize)]
        struct Helper {
            root: ObjectMeta,
            ctx: TxContext,
            action: VerifiedMoveAction,
            #[serde(default)]
            pre_execute_functions: Vec<FunctionCall>,
            #[serde(default)]
            post_execute_functions: Vec<FunctionCall>,
        }

        let helper = Helper::deserialize(deserializer)?;
        Ok(VerifiedMoveOSTransaction {
            root: helper.root,
            ctx: helper.ctx,
            action: helper.action,
        })
    }
}
```

[the remark this answers is not in the document]
Comment (jolestar) [reply moveos/moveos-types/src/transaction.rs:303]: answer from claude.ai

[the remark this answers is not in the document]
Comment (baichuan3) [reply moveos/moveos-types/src/transaction.rs:303]: https://github.com/rooch-network/rooch/pull/2310  This PR encounters a newly added field, and needs to ensure the compatibility of new and old data. There are two solutions, but there are also some problems that need to be solved

**Comment by baichuan3** [comment]
Try to implement it like this, it should work

```
#[derive(Debug, Clone)]
pub struct VerifiedMoveOSTransaction {
    pub root: ObjectMeta,
    pub ctx: TxContext,
    pub action: VerifiedMoveAction,
}


#[derive(Clone, Debug, Serialize, Deserialize)]
struct VerifiedMoveOSTransactionWrapper {
     root: ObjectMeta,
     ctx: TxContext,
     action: VerifiedMoveAction,
     #[serde(default)]
     pre_execute_functions: Vec<FunctionCall>,
     #[serde(default)]
     post_execute_functions: Vec<FunctionCall>,
}

impl Serialize forVerifiedMoveOSTransaction {
    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
    where
        S: Serializer,
    {
        let wrapper = VerifiedMoveOSTransactionWrapper {
            root: self.root.clone(),
            ctx: self.ctx.clone(),
            action: self.action.clone(),
            pre_execute_functions: vec![],
            post_execute_functions: vec![],
        };

        wrapper.serialize(serializer)
    }
}

impl<'de> Deserialize<'de> for VerifiedMoveOSTransaction {
    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
    where
        D: Deserializer<'de>,
    {
        let helper =VerifiedMoveOSTransactionWrapper::deserialize(deserializer)?;
        Ok(VerifiedMoveOSTransaction {
            root: helper.root,
            ctx: helper.ctx,
            action: helper.action,
        })
    }
}
```

**Review by jolestar — APPROVED** [review]

**Pull request merged**